### PR TITLE
Mrc 2033 Bug: hint cannot recover model run when hintr is torn down

### PR DESCRIPTION
**Pull request by carbowizzy**
## Description

Please describe your changes here

## Type of version change
_Delete as appropriate_

Patches

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/src/app/store/modelRun/mutations.ts [hunk 1]
```diff
@@ -51,8 +51,7 @@ export const mutations: MutationTree<ModelRunState> = {
     },
 
     [ModelRunMutation.RunStatusError](state: ModelRunState, action: PayloadWithType<Error>) {
-        state.errors.push(action.payload);
-        if (state.errors.length >= maxPollErrors) {
+        if (state.pollingCounter >= maxPollErrors) {
```
Comment (EmmaLRussell) [inline src/app/static/src/app/store/modelRun/mutations.ts:54]: I think we also need to reset pollingCounter as well don't we, for next time we want to run. Should do that in stopPolling, so it happens for all cases.

File: src/app/static/src/app/store/modelRun/mutations.ts [hunk 2]
```diff
@@ -61,6 +60,7 @@ export const mutations: MutationTree<ModelRunState> = {
                 detail: null
             });
         }
+        state.pollingCounter += 1
```
Comment (EmmaLRussell) [inline src/app/static/src/app/store/modelRun/mutations.ts:63]: Should be an `else` here.

File: src/app/static/src/tests/modelRun/mutations.test.ts [hunk 3]
```diff
@@ -98,12 +91,11 @@ describe("Model run mutations", () => {
...
             status: {done: false} as ModelStatusResponse,
-            modelRunId: "123"
+            modelRunId: "123",
+            pollingCounter: 9
         });
-
         mutations.RunStatusError(testState, {payload: "1"});
-
-        expect(testState.errors).toStrictEqual(["1"]);
+        expect(testState.errors).toStrictEqual([]);
```
Comment (EmmaLRussell) [inline src/app/static/src/tests/modelRun/mutations.test.ts:98]: This test should be renamed - no longer adds errors if if max not exceeded.

**Review by EmmaLRussell — CHANGES_REQUESTED** [review]

File: src/app/static/src/app/store/modelRun/mutations.ts (hunk 2 above)
Comment (carbowizzy) [reply src/app/static/src/app/store/modelRun/mutations.ts]: There's no need for else if the counter is above.

File: src/app/static/src/app/store/modelRun/mutations.ts [hunk 4]
```diff
@@ -51,6 +51,7 @@ export const mutations: MutationTree<ModelRunState> = {
     },
 
     [ModelRunMutation.RunStatusError](state: ModelRunState, action: PayloadWithType<Error>) {
+        state.pollingCounter += 1
```
Comment (EmmaLRussell) [inline src/app/static/src/app/store/modelRun/mutations.ts:54]: Of course, incrementing the poll here means you'll hit the limit one iteration earlier. I guess that's ok at this sort of scale.

File: src/app/static/src/tests/modelRun/mutations.test.ts [hunk 5]
```diff
@@ -66,7 +66,7 @@ describe("Model run mutations", () => {
         expect(testState.errors).toStrictEqual(["Test Error"]);
     });
 
-    it("run status error adds error and stops polling if max errors exceeded", () => {
+    it("run status error and stops polling if maxPollErrors exceeded", () => {
```
Comment (EmmaLRussell) [inline src/app/static/src/tests/modelRun/mutations.test.ts:69]: This one didn't need to be renamed as the error is added!

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**